### DISCUSSION TITLE
Upgrade to Redactor X

### DISCUSSION THREAD

**Discussion by mattcdavis1** [craftcms/redactor#399] [Ideas] [answered]
### Description

Not sure if [Redactor X](https://imperavi.com/redactorx/) would be an upgrade to this plugin or a separate plugin but it would be great to have this as an available option one way or another.

**Comment by brianrivet-tilt** [comment]
seconded!

**Comment by andris-sevcenko** [comment]
There would be a lot of work involved in this and it would probably have to be a different plugin.

**Comment by andris-sevcenko** [comment]
Well, I'll keep this open as not to lose this idea. But that would still most likely be a different plugin.

**Comment by uandco** [comment]
Since Imperavi [will stop supporting the "classic" Redactor in 2023](https://imperavi.com/redactor/), what is the roadmap to get Redactor X in Craft?

**Reply by eddiedale** [reply]
This feels like a pretty important topic, no?

I mean, one can't expect Craft to deliver a first-party plugin on Redactor X, but since Redactor is basically unsupported as of now, the question remains if this current first-party plugin will at all be supported going forward?

Starting new projects now in 2023, it seems like a potential risk to rely on this plugin without knowing the roadmap. 
At the same time it is by far the most popular plugin of all Craft CMS plugins.

Am I overthinking this?

**Comment by sandrodunkel** [comment]
Is there any news on this? Starting to get anxious using redactor on new projects…

**Reply by eddiedale** [reply]
I would definitely recommend using CKEditor going forward. It is actively maintained, and has gotten really good with the new releases. I'm pretty sure P&T has said clearly that Redactor won't be upgraded as a first-party plugin to Redactor X.

**Reply by sandrodunkel** [reply]
Ok then, guess I'll change to CKEditor. By any chance, do you know what exactly Imperavi means by "We will support classic Redactor until 2023"? Does this mean that they no longer roll out updates to the editor, or does that basically mean that the Craft Redactor plugin will stop working by the end of 2023?

**Reply by eddiedale** [reply]
No, I don't know exactly what they mean by that, but it definitely doesn't mean that the plugin will stop working by the end of 23. 
They basically have said "until 2023" anyway, so its already past.

The plugin will keep working, probably for years on, but it won't be actively pushed forward by P&T. But they will surely provide a fix if something critical pops up, seing the amount of installs..  I wouldn't call it risky to use on new projects, but it doesn't feel like the best decision.

**Comment by angrybrad** [comment, marked as answer]
@sandrodunkel 

> "We will support classic Redactor until 2023"?

Realistically, the classic Redactor hasn’t been supported for a few years by Imperavi. At least, based on the bug reports we’ve sent with little results.

Would highly recommend migrating to CKEditor as that’s where all of our development effort is going.

https://craftcms.com/blog/ckeditor-revamped

**Reply by sandrodunkel** [reply]
Alright, thanks Brad! How are you planning on communicating this switch to the community? Is the migration redactor->cke part of the Craft 5.0 migration doc? I assume most of the 80k redactor installs wont read this thread or the Craft Blog.

**Reply by angrybrad** [reply]
>  Is the migration redactor->cke part of the Craft 5.0 migration doc?

Nah, you can do that now on Craft 4.x : https://github.com/craftcms/ckeditor#converting-redactor-fields

We'll discuss how to communicate it more broadly.
